analysis/metrics.py

```diff
@@ -308,8 +314,8 @@ class MetricUtility:
 class PrivatePartitionSelectionUtility:
     """Stores aggregate metrics about partition selection."""
 
-    strategy: pipeline_dp.PartitionSelectionStrategy
-    num_partitions: float
+    strategy: Optional[pipeline_dp.PartitionSelectionStrategy]
```

Comment:
Why is this optional?

Reply:
technical reason, when it's created strategy isn't available. Having something optional is not a big problem, e.g. optional is recommended for all fields in protobuffers